src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/cluster/WorkflowBalancer.scala

```diff
@@ -65,19 +64,20 @@ class WorkflowBalancer @Inject() (
       workflows
     }
 
-  def resetSchedulerInstanceId(): Unit =
+  def resetSchedulerInstanceId(): Unit = {
+    logger.trace("Resetting scheduler instance id (SchedulerId={}) -> None", schedulerInstanceId)
     schedulerInstanceId = None
+  }
 
   private def getOrCreateInstance()(implicit ec: ExecutionContext) =
     schedulerInstanceId match {
       case Some(id) => Future(id)
       case None =>
         schedulerInstanceService
           .registerNewInstance()
-          .map { id =>
+          .map(wireTap { id =>
```

Comment:
Ok, I will remove it. But, IMO, this looks clearer when you see that there is no change in value, just execution of side-effect.

Akka had something similar in their stream. Even native scala contains `.tap`, or `.tapEach`. And it could be easily implemented on any `Functor` type with `cats`.
```scala
import cats._

implicit class FunctorOps[F[_], A](val fa: F[A]) extends AnyVal {
  def tap[A, U](f: A => U)(implicit func: Functor[F]): F[A] =
    func.map(fa) { elm =>
      f(elm)
      elm
    }
}
```

Because, to me, this looks much cleaner
```scala
schedulerInstanceId match {
  case Some(id) => Future.successful(id)
  case _        => 
    schedulerInstanceService
      .registerNewInstance()
      .tap(id => schedulerInstanceId = Some(id))
      .omComplete {
        case Success(id) =>
          logger.info(s"Successfully assigned new (SchedulerId=${id}) to scheduler")
        case Failure(e) =>
          logger.error("Failed to get new SchedulerId", e)
      }
}
```